service-provider.md

```diff
@@ -3,4 +3,113 @@ description: Service Bindings for Service Providers
 permalink: /service-provider/
 ---
 
-# TODO
+Service providers expose bindings through a `Secret` resource with data required for connectivity.  The specification's [Provisioned Service section][provisioned-service] describes this in detail.  Alternatively, the specification also supports [Direct Secret Reference][direct-secret-reference].  The only requirement for Direct Secret Reference is a `Secret` resource with data required for connectivity.  Alternatively, if you are creating a Provisioned Service (preferred approach), you also need a custom resource with `.status.binding.name` attribute pointing to the `Secret` resource.
+
+Here is an example custom resource:
+
+```yaml
+apiVersion: example.dev/v1beta1
+kind: Database
+metadata:
+  name: database-service
+...
+status:
+  binding:
+    name: production-db-secret
+```
+
+
+{% include note.html content="Implementing the Provisioned Service contract frees the `ServiceBinding` from needing to know about the name of the `Secret` holding the credentials.  The service can update the secret name exposed over time.  This behavior not only decouples the ServiceBinding and the workload from needing to know the name of the Secret, it also enables use of immutable secrets.  An immutable secret cannot be updated after it is created.  To rotate the credentials, a new secret can be created and update the same in the Provisioned Service resource's `.status.binding.name` attribute." %}
```

Comment:
nit: The ServiceBinding needs to know the name of the secret, but the author of that resource doesn't

```suggestion
{% include note.html content="Implementing the Provisioned Service contract frees the `ServiceBinding` creator from needing to know about the name of the `Secret` holding the credentials.  The service can update the secret name exposed over time.  This behavior not only decouples the ServiceBinding and the workload from needing to know the name of the Secret, it also enables use of immutable secrets.  An immutable secret cannot be updated after it is created.  To rotate the credentials, a new secret can be created and update the same in the Provisioned Service resource's `.status.binding.name` attribute." %}
```